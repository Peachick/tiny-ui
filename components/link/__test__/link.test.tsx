import React from 'react';
import { render } from '@testing-library/react';
import Link from '../index';

describe('<Link />', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Link href="#">Link</Link>);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('external', () => {
    it('should set _blank to target attribute by default', () => {
      const { container } = render(<Link>Link</Link>);
      expect(container.firstChild).toHaveAttribute('target', '_blank');
    });

    it('should set _self to target attribute by default', () => {
      const { container } = render(<Link external={false}>Link</Link>);
      expect(container.firstChild).toHaveAttribute('target', '_self');
    });
  });

  it('should be disabled', () => {
    const { container } = render(<Link disabled>Link</Link>);
    expect(container.firstChild).toHaveClass('ty-link_disabled');
  });
});
