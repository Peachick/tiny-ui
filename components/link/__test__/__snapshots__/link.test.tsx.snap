// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should match the snapshot 1`] = `
<DocumentFragment>
  <a
    aria-disabled="false"
    class="ty-link"
    href="#"
    role="link"
    target="_blank"
  >
    <span>
      Link
    </span>
  </a>
</DocumentFragment>
`;
